platform/android/src/style/sources/geojson_source.cpp

```diff
@@ -147,5 +176,37 @@ namespace android {
         );
     }
 
+    void FeatureConverter::convertJson(std::string json,
+                                       ActorRef<Callback> callback) {
+        using namespace mbgl::style::conversion;
+
+        android::UniqueEnv _env = android::AttachEnv();
+
+        // Convert the jni object
+        Error error;
+        optional<GeoJSON> converted = parseGeoJSON(json, error);
+        if(!converted) {
+            mbgl::Log::Error(mbgl::Event::JNI, "Error setting geo json: " + error.message);
+            return;
+        }
+
+        callback.invoke(&Callback::operator(), *converted);
+    }
+
+    template<class JNIType>
+    void FeatureConverter::convertObject(jni::Object<JNIType> jObject, ActorRef<Callback> callback) {
+        using namespace mbgl::android::geojson;
+
+        android::UniqueEnv _env = android::AttachEnv();
+        // Convert the jni object
+        auto geometry = JNIType::convert(*_env, jObject);
+        callback.invoke(&Callback::operator(), GeoJSON(geometry));
+    }
+
+    Update::Update(Converter _converterFn, std::unique_ptr<Actor<Callback>> _callback) {
```

Comment:
Since this is a class, you can do the initialization as
```
... )
, converterFn(_converterFn)
, callback(std::move(_callback)
{}
```

This is mostly just a matter of style although it can save a tiny bit of work (with the assignment in the body of the constructor, you have to default-initialize the values first and _then_ assign them).